pkg/oc/cli/admin/release/image_mapper_test.go

```diff
@@ -255,3 +256,104 @@ func TestNewExactMapper(t *testing.T) {
 		})
 	}
 }
+
+func TestNewComponentVersionsMapper(t *testing.T) {
+	type args struct {
+	}
+	tests := []struct {
+		name        string
+		releaseName string
+		versions    map[string]string
+		imagesByTag map[string][]string
+		in          string
+		out         string
+		wantErr     string
+	}{
+		{
+			in:  `version: 0.0.1-snapshot\n`,
+			out: `version: 0.0.1-snapshot\n`,
+		},
+		{
+			in:      `version: 0.0.1-snapshot-\n`,
+			wantErr: `unknown version reference ""`,
+		},
+		{
+			in:      `version: 0.0.1-snapshot-a\n`,
+			wantErr: `unknown version reference "a"`,
+		},
+		{
+			releaseName: "2.0.0",
+			in:          `version: 0.0.1-snapshot\n`,
+			out:         `version: 2.0.0\n`,
+		},
+		{
+			name:        "release name is not semver",
+			releaseName: "2.0",
+			in:          `version: 0.0.1-snapshot\n`,
+			out:         `version: 0.0.1-snapshot\n`,
+		},
+		{
+			versions: map[string]string{"a": "2.0.0"},
+			in:       `version: 0.0.1-snapshot-a\n`,
+			out:      `version: 2.0.0\n`,
+		},
+		{
+			versions:    map[string]string{"a": "2.0.0"},
+			imagesByTag: map[string][]string{"a": {"tag1", "tag2"}},
+			in:          `version: 0.0.1-snapshot-a\n`,
+			wantErr:     `the version for "a" is inconsistent across the referenced images: tag1, tag2`,
```

Comment:
include a test-case where there is inconsistency on `a`, but was not requested...